mobile/src/main/java/ch/epfl/sdp/mobile/application/chess/rules/Effect.kt

```diff
@@ -46,6 +46,17 @@ fun interface Effect<Piece : Any> {
       it.set(position, piece)
     }
 
+    /**
+     * Applies a sequence of [Effect] in order, from left to right. This may be useful if you want
+     * to combine some simple effects into a single action.
+     *
+     * @param P the type of the pieces.
+     * @param effects the sequence of effects that should be applied.
+     */
+    fun <P : Any> all(vararg effects: Effect<P>): Effect<P> = Effect {
```

Comment:
For me, the name `all`isn't clear enough, it doesn't reflect that the effect is applied one by one. I suggest this :
```suggestion
    fun <P : Any> chain(vararg effects: Effect<P>): Effect<P> = Effect {
```